ecr/ref.go

```diff
@@ -70,6 +74,60 @@ func (spec ECRSpec) Registry() string {
 	return spec.arn.AccountID
 }
 
+// ParseEcrImageNameToRef takes an ECR image name and returns its canonical `ref` format.
+// i.e. ecr.aws/arn:<partition>:ecr:<region>:<account>:repository/<name>:<tag>
+func ParseEcrImageNameToRef(input string) (string, error) {
```

Comment:
nit: I'd prefer it if this were in a different location in the file.  Either it's a constructor to the `ECRSpec` and should be right below `ParseRef`, or it's not really related to the rest of the functions and should probably be in its own file or at the bottom of this one.